SU-SWS/ansible-playbooks
# SITES-464: Draft of protecting prod sites

# READY FOR REVIEW

# Summary
- Adds a new role and dependencies (on add-vhost, change-paths, setup-site, and upload-site) that will fail any host on a list of production sites.  The list is stored in a separate private repository, ansible-sync.
- I've also removed the inventory directory from this repository and moved it into ansible-sync.

# Needed By (5/17)
- End of sprint

# Urgency
- Perhaps a 4 or 5 at the moment, as we do not have another launch in the immediate future.

# Steps to Test

1. Checkout this branch.
2. Clone a copy of https://github.com/SU-SWS/ansible-sync to the ansible-playbooks root directory.
3. Run: `ansible-playbook -i ansible-sync/sites.yml migration-playbook.yml`.
4. `eao` should fail and `TELOS` should continue.
5. `e-ao` shoudl fail with the message, "eao conflicts with an inventory hostname on our list of protected production sites."

After testing and before merging, we'll want to change the value of the environment from `test` to `prod`.

# Affected Projects or Products
- Sites 2.0

# Associated Issues and/or People
- Version control inventory: https://stanfordits.atlassian.net/browse/SITES-497
- Protect prod sites: https://stanfordits.atlassian.net/browse/SITES-464
